Automate basic scenarios related to exechost_periodic hook

Mahalty opened the pull request:
<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Describe Bug or Feature

- Automate scenarios of exechost_periodic hook


#### Describe Your Change

- Automate following scenarios

1. exechost_periodic hook that accepts the event
2. exechost_periodic hook that is interrupted by alarm call
3. exechost_periodic hook that causes unhandled exception
4. setting custom resource through exechost_periodic hook


#### Link to Design Doc
<!--- If there is a design, link to it here: **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)** -->


#### Attach Test and Valgrind Logs/Output

- [Exec_periodic.txt](https://github.com/openpbs/openpbs/files/6480617/Exec_periodic.txt)




<!--- Pull Request Guidelines: [Pull Request Guidelines](https://pbspro.atlassian.net/wiki/spaces/DG/pages/1187348483/Pull+Request+Guidelines) -->

bayucan reviewed (approved):
Looks good.

bayucan commented:
@Mahalty : you have some failing checks, though..

bhroam commented on test/tests/functional/pbs_hook_exechost_periodic.py:
```diff
@@ -74,3 +74,130 @@ def test_multiple_exechost_periodic_hooks(self):
+        """
+        hook_name = "periodic"
+        hook_attrs = {'event': 'exechost_periodic', 'enabled': 'True'}
+        hook_script = "x"
```
Instead of using a syntax error to raise an exception, why not raise one yourself?  It is clearer to the reader.

bhroam commented on test/tests/functional/pbs_hook_exechost_periodic.py:
```diff
@@ -74,3 +74,130 @@ def test_multiple_exechost_periodic_hooks(self):
+        remote_node = node.name
+        pbs.logmsg(pbs.LOG_DEBUG, "remote node is %s" % node.name)
+vn = pbs.event().vnode_list
+if remote_node not in vn.keys():
```
No need to say vn.keys().  saying 'if remote_node not vn' is the same.  It looks at the keys. (here and below)

bhroam commented on test/tests/functional/pbs_hook_exechost_periodic.py:
```diff
@@ -74,3 +74,130 @@ def test_multiple_exechost_periodic_hooks(self):
+        node_attribs = {'resources_available.foo': True}
+        self.server.expect(NODE, node_attribs, id=self.hostB)
+
+common_periodic_hook_script = """import pbs
```
Move this above the class (or even make it a class variable, but up top).  It allows the reader to see what the hook is before reading all the tests.

Mahalty commented:
@bhroam  I have addressed your review comments. Please review

Latest Execution-logs:
[Exec_periodic_1.txt](https://github.com/openpbs/openpbs/files/6493717/Exec_periodic_1.txt)

bhroam reviewed (approved):
LGTM

bayucan commented:
Merged #2391 into master.
